owncloudApp/src/main/java/com/owncloud/android/presentation/ui/sharing/ShareActivity.kt

```diff
@@ -0,0 +1,385 @@
+class ShareActivity : FileActivity(), ShareFragmentListener {
+    /**
+     * Shortcut to get access to the [ShareFileFragment] instance, if any
+     *
+     * @return A [ShareFileFragment] instance, or null
+     */
+    private val shareFileFragment: ShareFileFragment?
+        get() = supportFragmentManager.findFragmentByTag(TAG_SHARE_FRAGMENT) as ShareFileFragment
+
+    /**
+     * Shortcut to get access to the [PublicShareDialogFragment] instance, if any
+     *
+     * @return A [PublicShareDialogFragment] instance, or null
+     */
+    private val publicShareFragment: PublicShareDialogFragment?
+        get() = supportFragmentManager.findFragmentByTag(TAG_PUBLIC_SHARE_DIALOG_FRAGMENT) as PublicShareDialogFragment?
+
+    /**
+     * Shortcut to get access to the [EditPrivateShareFragment] instance, if any
+     *
+     * @return A [EditPrivateShareFragment] instance, or null
+     */
+    private val editPrivateShareFragment: EditPrivateShareFragment?
+        get() = supportFragmentManager.findFragmentByTag(TAG_EDIT_SHARE_FRAGMENT) as EditPrivateShareFragment?
+
+    // Needed for changing some UI elements in fragments before loading them, otherwise there will be UI glitches
+    private val ocCapabilityViewModel: OCCapabilityViewModel by viewModel {
+        parametersOf(
+            account
+        )
+    }
+
+    private val ocShareViewModel: OCShareViewModel by viewModel {
+        parametersOf(
+            file.remotePath,
+            account
+        )
+    }
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        setContentView(R.layout.share_activity)
+
+        // Set back button
+        supportActionBar?.setDisplayHomeAsUpEnabled(true)
+
+        val ft = supportFragmentManager.beginTransaction()
+
+        if (savedInstanceState == null && file != null && account != null) {
+            // Add Share fragment on first creation
+            val fragment = ShareFileFragment.newInstance(file, account!!)
+            ft.replace(
+                R.id.share_fragment_container, fragment,
+                TAG_SHARE_FRAGMENT
+            )
+            ft.commit()
+        }
+
+        observePrivateShareCreation()
+    }
+
+    public override fun onStop() {
+        super.onStop()
+    }
+
+    override fun observeCapabilities() {
+        ocCapabilityViewModel.capabilities.observe(
+            this,
+            Observer { uiResult ->
+                when (uiResult?.status) {
+                    Status.SUCCESS -> {
+                        if (publicShareFragment != null) {
+                            publicShareFragment?.updateCapabilities(uiResult.data)
+                        } else {
+                            shareFileFragment?.updateCapabilities(uiResult.data)
+                        }
+                        dismissLoadingDialog()
+                    }
+                    Status.ERROR -> {
+                        if (publicShareFragment != null) {
```

Comment:
you have a bug here. Follow these steps to reproduce it:

1. Open the application with Activity or Fragment
2. Cause an error, e.g. switch airplane mode on and tap a button that makes a request to server
3. Observe the error is shown
4. Rotate the device

Current behavior: Error is shown again and the user has not done any request.
Expected behavior: No error is shown

Reply:
Thanks, I have included this bug in the modularization epic (https://github.com/owncloud/android/issues/2607#issuecomment-550279458) , it's better to fix it when some of the coming PRs are already merged since some parts have been changed. Otherwise, we will change it twice.